tfhe/src/shortint/parameters/mod.rs

```diff
@@ -501,11 +505,17 @@ impl ShortintParameterSet {
         }
     }
 
-    pub const fn encryption_key_choice(&self) -> EncryptionKeyChoice {
+    pub fn encryption_key_choice(&self) -> EncryptionKeyChoice {
         match self.inner {
             ShortintParameterSetInner::PBSOnly(params) => params.encryption_key_choice(),
             ShortintParameterSetInner::WopbsOnly(params) => params.encryption_key_choice,
-            ShortintParameterSetInner::PBSAndWopbs(params, _) => params.encryption_key_choice(),
+            ShortintParameterSetInner::PBSAndWopbs(pbs_params, wop_pbs_params) => {
+                assert_eq!(
+                    pbs_params.encryption_key_choice(),
+                    wop_pbs_params.encryption_key_choice
+                );
+                pbs_params.encryption_key_choice()
+            }
```

Comment:
is this required ? If not I'd rather have it in a separate PR/issue as we keyswitch between those parameter sets and the parameter set we use for encryption is the one from the PBS, if you think this deserves a comment I'm fine with it but I don't think this change is correct (because we keyswitch from PBS to WOPBS and then back to PBS when required)

Reply:
Removed